Src/Core/Common/ExpressionExtensions.cs

```diff
@@ -129,9 +131,17 @@ public static class ExpressionExtensions
                         node = null;
                         break;
 
+                    case ExpressionType.Call:
+                        var methodCallExpression = node as MethodCallExpression;
```

Comment:
If the `NoteType` is `Call`, can the `node` be anything else than a `MethodCallExpression`? Because the `as` operator does kind of imply that, as well as the `?` operator when you try to get the `Method`.